src/data_hub/dataframe.py

```diff
@@ -0,0 +1,159 @@
+"""
+This module defines a class called DataFrame that is a drop-in
+replacement for a pandas DataFrame, but also allows automatic
+updating of Deep Origin databases.
+"""
+
+from datetime import datetime
+from typing import Optional
+
+import humanize
+import pandas as pd
+from deeporigin.data_hub import api
+from deeporigin.utils import construct_resource_url
+
+
+class DataFrame(pd.DataFrame):
```

Comment:
For consistency with pandas, I think we could add a `from_deeporigin` method rather than encouraging users to use `api.get_data_frame`. 

`from_deeporigin` could be implementing using `api.get_data_frame`. It could largely be syntactic sugar for `api.get_data_frame`.

Reply:
I like the mirroring of pandas functions. Pandas has pd.read_csv, maybe we could have deeporigin.read_database?

Reply:
The DataFrame class includes `from_dict` and `from_records`.  It seems more straightforward to add methods to the DataFrame class. In that case, I think we should follow the pattern `from_*`.

`read_*` are on the pandas module.

Reply:
i agree.

pandas is weird here -- we're supposed to use `pd.read_csv` and not `pd.DataFrame.from_csv` [link](https://pandas.pydata.org/pandas-docs/version/0.23.4/generated/pandas.DataFrame.from_csv.html)

but we're supposed to use `pd.DataFrame.from_dict` and not `pd.from_dict`

Reply:
fixed